### ISSUE TITLE
Support 64bit integer properly

### ISSUE THREAD

**Issue by xylosper** [mapbox/wagyu#104]
The example code utilizes 64bit integer, but it does not support 64bit integer actually.

For instance, slope comparison is done with multiplication of two coordinate delta and therefore, one if each coordinate value exceeds 32bit range, it causes overflow.
Original Clipper utilizes 128bit integer operation to compare slope.

**Comment by flippmoke** [comment]
Yes, there are several parts of the code that would need to be updated to support this properly and you are absolutely correct. I do not see myself having time to add support for this, but definitely willing to support reviewing a PR for it!
